.github/workflows/docs.yml

```diff
@@ -25,56 +18,45 @@ jobs:
-        id: gen
-        shell: bash
-        run: |
-          # Create temp dir to hold docs
-          DOCS_DIR=$(mktemp -d)
-          echo "dir=${DOCS_DIR}" >> $GITHUB_OUTPUT
-          juju documentation --split --out=$DOCS_DIR --discourse-ids $TOPIC_IDS
-        # TODO: save $DOCS_DIR as an artifact
-
-      - name: Generate controller config keys doc
-        env:
-          DOCS_DIR: ${{ steps.gen.outputs.dir }}
-          JUJU_SRC_ROOT: '.'
-        run: |
-          go run ./scripts/md-gen/controller-config
-
-      - name: Generate hook tool docs
-        env:
-          DOCS_DIR: ${{ steps.gen.outputs.dir }}
-        run: |
-          go run ./scripts/md-gen/hook-tools
-
-      - name: Add version notice to generated docs
-        env:
-          DOCS_DIR: ${{ steps.gen.outputs.dir }}
-        run: |
-          go run ./scripts/md-gen/version-notice
-
-      - name: pip install requirements
-        run: |
-          python3 -m venv .venv
-          .venv/bin/python3 -m pip install -r ./scripts/discourse-sync/requirements.txt
-
-      - name: Check docs on Discourse
-        if: github.event_name == 'pull_request'
-        env:
-          DOCS_DIR: ${{ steps.gen.outputs.dir }}
-        run: |
-          .venv/bin/python3 ./scripts/discourse-sync/main.py check
-
-      - name: Sync docs to Discourse
-        if: github.event_name == 'push'
-        env:
-          DISCOURSE_API_USERNAME: ${{ secrets.DISCOURSE_API_USERNAME }}
-          DISCOURSE_API_KEY: ${{ secrets.DISCOURSE_API_KEY }}
-          DOCS_DIR: ${{ steps.gen.outputs.dir }}
-        run: |
-          .venv/bin/python3 ./scripts/discourse-sync/main.py sync
+      - name: Install sphinx dependencies
+        run: |
+          sudo DEBIAN_FRONTEND=noninteractive apt -y update
+          sudo DEBIAN_FRONTEND=noninteractive apt -y install python3.10-venv
+          python -m venv venv
+          source venv/bin/activate
+          python -m pip install --upgrade --no-cache-dir pip setuptools
+          python -m pip install --exists-action=w --no-cache-dir -r docs/.sphinx/requirements.txt
+
+      - name: Make documentation
+        run: |
+          make docs-html
+
+      - name: Check for changes
+        run: |
+          git add -A
+          git rm -rf --quiet --cached venv/
+          if [[ -n $(git diff HEAD) ]]; then
+            # Print the full diff for debugging purposes
+            git diff HEAD
+            echo "*****"
+            echo "The following generated documentation has been modified:"
+            git diff --name-status HEAD
+            echo "Please regenerate these files and check in the changes."
+            echo ""
+            echo "When running locally, the CLI documentation is generated by the version of juju on your path, NOT the version in the tree"
+            echo "To regenerate the documentation locally, install the juju CLI from your local tree and build the docs with:"
+            echo "    go install -v -tags minimal ./cmd/juju"
```

Comment:
This isn't quite right, fix it here and we can merge forward

Simply `make juju docs-html`

Reply:
Changed